### PR TITLE
⚠️ Upgrade cert-manager to v1.1.0

### PR DESCRIPTION
**Pull request by fabriziopandini**
**What this PR does / why we need it**:
This PR updates the version of cert-manager from v0.16.1 to v1.1.0

Waiting for https://github.com/kubernetes-sigs/cluster-api/pull/3985 to change manifests

**Which issue(s) this PR fixes**:
Fixes https://github.com/kubernetes-sigs/cluster-api/issues/3996

### CODE REVIEW

**Comment by vincepri** [comment]
@fabriziopandini What's left to do here? Could we also bring these changes to Kubebuilder?

**Comment by fabriziopandini** [comment]
@vincepri from what I understood in kubebuilder this was already addressed by https://github.com/kubernetes-sigs/kubebuilder/pull/1840

This PR is waiting for changes to webhook to land (#3985) before migrating certificates to the v1 version

**Comment by wfernandes** [comment]
Reviewing 👀

File: cmd/clusterctl/client/cluster/cert_manager.go [hunk 1]
```diff
@@ -101,30 +101,22 @@ func (cm *certManagerClient) setManifestHash() error {
...
 	if err != nil {
 		return err
 	}
-	found := false
 
-	for i := range objs {
-		o := objs[i]
-		if o.GetKind() == "CustomResourceDefinition" {
-			labels := o.GetLabels()
-			version, ok := labels[certmanagerVersionLabel]
```
Comment (wfernandes) [inline cmd/clusterctl/client/cluster/cert_manager.go:115]: We should remove the const `certmanagerVersionLabel = "helm.sh/chart"` from line 54 above.

File: docs/book/src/developer/providers/v1alpha3-to-v1alpha4.md [hunk 2]
```diff
@@ -41,3 +41,7 @@ the delegating client by default under the hood, so this can be now removed.
 - The functions `fake.NewFakeClientWithScheme` and `fake.NewFakeClient` have been deprecated.
 - Switch to `fake.NewClientBuilder().WithObjects().Build()` instead, which provides a cleaner interface
   to create a new fake client with objects, lists, or a scheme.
+
+## Upgrade cert-manager to v1.1.0
+
+- TODO
```
Comment (wfernandes) [inline docs/book/src/developer/providers/v1alpha3-to-v1alpha4.md:47]: I'm guessing this is part of the WIP.

**Review by wfernandes — COMMENTED** [review]
We should probably also update our documentation.
https://cluster-api.sigs.k8s.io/developer/guide.html?highlight=cert-manage#cert-manager
In the link above, we are specifying `v0.11.0` instead of `v0.16.1`. This PR can update that to `v1.1.0`.

**Comment by vincepri** [comment]
@fabriziopandini `make generate` needs to be updated

**Review by vincepri — APPROVED** [review]
/approve
/lgtm

**Pull request merged**